File: app/models/qdm/identifier.rb
module QDM
  # app/models/qdm/identifier.rb
  class Identifier
    include Mongoid::Document
    embedded_in :data_element
    field :namingSystem, type: String
    field :value, type: String
    field :qdmVersion, type: String, default: '5.5'
  end
end

File: app/assets/javascripts/attributes/Identifier.js
const mongoose = require('mongoose/browser');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const IdentifierSchema = mongoose.Schema({
  namingSystem: String,
  value: String,
  qdmVersion: { type: String, default: '5.6' },
  _type: { type: String, default: 'QDM::Identifier' },

}, { _id: false, id: false });

module.exports.IdentifierSchema = IdentifierSchema;
class Identifier extends mongoose.Document {
  constructor(object) {
    super(object, IdentifierSchema);
    this._type = 'QDM::Identifier';
  }
}
module.exports.Identifier = Identifier;
